From ba36bfe25f9381e3f12c0281c20eab738475bba8 Mon Sep 17 00:00:00 2001
From: Andrea Canciani <ranma42@gmail.com>
Date: Sat, 25 Feb 2012 23:11:52 +0100
Subject: [PATCH 2/5] imm32: Avoid deadlock when destroying thread data (v2)

When thread-local data is being freed, avoid sending synchronous
messages to other threads as that can result in deadlocks if both
threads are terminating.
---
 dlls/imm32/imm.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git wine/dlls/imm32/imm.c b/dlls/imm32/imm.c
index 7394101..37553d1 100644
--- wine/dlls/imm32/imm.c
+++ wine/dlls/imm32/imm.c
@@ -676,7 +676,7 @@ static BOOL IMM_DestroyContext(HIMC hIMC)
     {
         data->immKbd->uSelected --;
         data->immKbd->pImeSelect(hIMC, FALSE);
-        SendMessageW(data->IMC.hWnd, WM_IME_SELECT, FALSE, (LPARAM)GetKeyboardLayout(0));
+        SendNotifyMessageW(data->IMC.hWnd, WM_IME_SELECT, FALSE, (LPARAM)GetKeyboardLayout(0));
 
         if (IMM_GetThreadData()->hwndDefault == data->imeWnd)
             IMM_GetThreadData()->hwndDefault = NULL;
-- 
1.7.7.5 (Apple Git-26)

